tests/AnalyticsEventReporter.test.ts

```diff
@@ -1,34 +1,217 @@
+      });
+  });
+
+  it('should call postWithFetch with correct fields, ' +
+      'report should return success json if post returns successful',
+  async () => {
+    mockPostWithFetch.mockResolvedValue({
+      ok: true,
+      status: 202,
+      statusText: 'OK',
+      json: jest.fn().mockResolvedValue({id: 1111}),
+      redirected: false,
+      type: 'basic',
+      url: 'https://example.com',
+      clone: jest.fn(),
+      headers: new Headers(),
+      body: null,
+      bodyUsed: false,
+      arrayBuffer: jest.fn(),
+      blob: jest.fn(),
+      formData: jest.fn(),
+      text: jest.fn()
+    });
+    mockUseBeacon.mockReturnValueOnce(false);
+
+    const config: AnalyticsConfig = {
+      bearer: 'bearerToken',
+      env: EnvironmentEnum.Sandbox,
+    };
+    const reporter = new AnalyticsEventReporter(config).with({
+      action: 'ADD_TO_CART',
+      referrerUrl: 'https://yext.com',
+      count: 5,
+    });
+
+    const res = await reporter.report({
+      destinationUrl: 'https://google.com',
+      referrerUrl: null,
+    });
+
+    // Expect Successful Response
+    expect(res).toEqual({id: 1111});
+    /** Expect merge to have completed correctly (with referrerUrl being removed),
+       * the url to be constructed correctly defaulting to US,
+       * and the clientSdk and authorization to be added to the request body in the correct format. **/
+    expect(mockPostWithFetch).toHaveBeenCalledWith(
+      'https://sbx.us.yextevents.com/accounts/me/events',
+      {
+        action: 'ADD_TO_CART',
+        authorization: 'Bearer bearerToken',
+        clientSdk: {
+          '@yext/analytics': '1.0.0-beta.0'
+        },
+        destinationUrl: 'https://google.com',
+        count: 5,
+      });
+  });
+
+  it('should call post with correct fields, report should return error json if post returns an error',
+    async () => {
+      const mockSetupSessionId = getOrSetupSessionId as jest.MockedFunction<typeof getOrSetupSessionId>;
+
+      mockSetupSessionId.mockImplementation( () => 'ULID1234');
+
+      mockPostWithFetch.mockResolvedValue({
+        ok: true,
+        status: 401,
+        statusText: 'Unauthorized request',
+        json: jest.fn().mockResolvedValue({id: 1111, errors: ['Unauthorized request']}),
+        redirected: false,
+        type: 'basic',
+        url: 'https://example.com',
+        clone: jest.fn(),
+        headers: new Headers(),
+        body: null,
+        bodyUsed: false,
+        arrayBuffer: jest.fn(),
+        blob: jest.fn(),
+        formData: jest.fn(),
+        text: jest.fn()
+      });
+      mockUseBeacon.mockReturnValueOnce(false);
+
+      const config: AnalyticsConfig = {
+        bearer: 'bearerToken',
+        sessionTrackingEnabled: true,
+      };
+      const reporter = new AnalyticsEventReporter(config).with({
+        action: 'ADD_TO_CART',
+        referrerUrl: 'https://yext.com',
+        count: 5,
+      });
+
+      const res = await reporter.report({
```

Comment:
Can we add the following tests:

- calling report with no argument
- Create then calling report with argument (no with() call)
- create then calling report without argument (no with() call)
- create and call with() more than once, and verifying that the two different Service objects are called with different values.